fraud/service.go

```diff
@@ -0,0 +1,147 @@
+package fraud
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+const (
+	fetchHeaderTimeout = time.Minute * 2
```

Comment:
I know this is private but I still think this deserves a comment. Ideally, what the rationale behind the 2 mins is as well.

Reply:
comment added.